pkg/utils/nodepool/nodepool.go

```diff
@@ -132,3 +135,38 @@ func PatchStatus(ctx context.Context, c client.Client, stored, nodePool *v1beta1
 func HashAnnotation(nodePool *v1beta1.NodePool) map[string]string {
 	return map[string]string{v1beta1.NodePoolHashAnnotationKey: nodePool.Hash()}
 }
+
+func UpdateStatusCondition(ctx context.Context, c client.Client, np *v1beta1.NodePool, conditionType v1beta1.NodePoolConditionType, condition v1.ConditionStatus) {
```

Comment:
A lot of these functions are already shipped as part of `knative.dev/pkg`. We should just use the same logic that we are using elsewhere in the codebase. See things like `MarkTrue` that's present in some of the NodeClaim lifecycle controllers.